### PR TITLE
Add message to gitignore dialog when not on the default branch

### LINKED ISSUE #3720: Update `gitignore` dialogue if not on default branch
Add differentiating text to the `gitignore` dialogue if the user is not on the default branch. 

As each branch can have its own `gitignore`, the Repo Settings > Ignored Files text should have a subtle confirmation to the user that they are saving a `gitignore` file only for that branch.

screen shot 2018-01-03 at 10 38 19 am

https://github.com/desktop/desktop/blob/a61a5bdc94ee8237dfff328957cdaee99a9b61e1/app/src/ui/repository-settings/git-ignore.tsx#L16

OSX, 1.0.12-beta0

cc @lee-dohm

### PR DESCRIPTION
**Pull request by lee-dohm**
Fixes #3720

I'm still not super confident on the language in the dialog but this is the best I've come up with so far. Thoughts?

/cc @shiftkey @tierninho

### CODE REVIEW

File: app/src/ui/repository-settings/git-ignore.tsx [hunk 1]
```diff
@@ -23,6 +37,9 @@ export class GitIgnore extends React.Component<IGitIgnoreProps, {}> {
           for more information about the file format, or simply ignore a file by
           right clicking on it in the uncommitted changes view.
         </p>
+
+        {this.renderBranchWarning()}
```
Comment (j-f1) [inline app/src/ui/repository-settings/git-ignore.tsx]: `{this.props.isDefaultBranch && <p>...</p>}`?

Comment (lee-dohm) [reply app/src/ui/repository-settings/git-ignore.tsx]: That would return `false` instead of `null` in the negative case. Would that render differently?

Comment (j-f1) [reply app/src/ui/repository-settings/git-ignore.tsx]: [Nope](https://reactjs.org/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored):

> `false`, `null`, `undefined`, and `true` are valid children. They simply don’t render.

Comment (lee-dohm) [reply app/src/ui/repository-settings/git-ignore.tsx]: Thanks for that @j-f1. Still, six of one, half dozen of the other. I feel this looks cleaner and the same pattern is used in other UI elements.

**Comment by tierninho** [comment]
@lee-dohm language looks good to me. Slight tweak?

Current:
You are not on the default branch, so changes here may not be applied to the repository when you switch branches.

New?:
As you are not on the default branch, changes here may not be applied to the repository when you switch branches.

**Comment by lee-dohm** [comment]
Thanks @tierninho. Updated :+1:

**Comment by lee-dohm** [comment]
@niik I totally understand your concerns. I'll take a stab at it. I like the wording you include in your screenshots, I'll see if I can expand on it just a little bit.

**Comment by lee-dohm** [comment]
Here's the updated view:

screen shot 2018-03-05 at 9 00 51 am

**Comment by niik** [comment]
@lee-dohm I think that's an improvement but I worry that the 'per-branch' phrasing may be confusing, that it could be interpreted as gitignore files somehow being attached to branches. I don't have a great alternative though 😕

**Comment by lee-dohm** [comment]
I can take out the "per-branch" but leave the improved wording. We still need to decide what to do with #3720 though. Should we just punt on it for now by closing it and mark it for the future? Or leave it open?

**Comment by shiftkey** [comment]
@lee-dohm @niik I'm fine with closing out #3720 if we don't think there's value in talking about "per-branch" - @tierninho any objections given the discussion above?

**Comment by niik** [comment]
> @lee-dohm @niik I'm fine with closing out #3720 if we don't think there's value in talking about "per-branch" here

I'm happy with that. @lee-dohm if you have time please pull that per-branch phrasing and then we'll be ready to land this once @tierninho has weighed in.

**Comment by lee-dohm** [comment]
Done.

**Comment by niik** [comment]
> I'm happy with that. @lee-dohm if you have time please pull that per-branch phrasing and then we'll be ready to land this once @tierninho has weighed in.

I just realized that @tierninho is off today so I'll just bring this in and we can continue to iterate when he gets back.

**Comment by lee-dohm** [comment]
Awesome, thanks @niik :+1:

**Pull request merged**